unikraft/app/loader.go

```diff
@@ -35,6 +35,13 @@ const (
 func NewApplicationFromInterface(ctx context.Context, iface map[string]interface{}, popts *ProjectOptions) (Application, error) {
 	app := application{}
 
+	if l, ok := iface["labels"]; ok {
+		app.labels, ok = l.(map[string]string)
+		if !ok {
+			return nil, errors.New("labels must be a map")
+		}
+	}
+
```

Comment:
This crashes for me depending on what I place in the map. Let's use an existing transformer.

```diff
diff --git a/unikraft/app/loader.go b/unikraft/app/loader.go
index 933d4ed6..26677ff7 100644
--- a/unikraft/app/loader.go
+++ b/unikraft/app/loader.go
@@ -35,13 +35,6 @@ const (
 func NewApplicationFromInterface(ctx context.Context, iface map[string]interface{}, popts *ProjectOptions) (Application, error) {
        app := application{}
 
-       if l, ok := iface["labels"]; ok {
-               app.labels, ok = l.(map[string]string)
-               if !ok {
-                       return nil, errors.New("labels must be a map")
-               }
-       }
-
        name := ""
        if n, ok := iface["name"]; ok {
                name, ok = n.(string)
@@ -111,6 +104,10 @@ func NewApplicationFromInterface(ctx context.Context, iface map[string]interface
                return nil, err
        }
 
+       if err := Transform(ctx, getSection(iface, "labels"), &app.labels); err != nil {
+               return nil, err
+       }
+
        extensions := getSectionMap(iface, "extensions")
        if len(extensions) > 0 {
                app.extensions = extensions
```

Reply:
I had doubts this would work, turns out I should have tested :D